src/main/java/wallet/model/contact/ContactList.java

```diff
@@ -0,0 +1,146 @@
+package wallet.model.contact;
+
+import java.util.ArrayList;
+
+public class ContactList {
+    private boolean isModified = false;
+    /**
+     * Stores the current list of contacts of the user.
+     */
+    private ArrayList<Contact> contactList;
+
+    /**
+     * Constructs a new ContactList object.
+     */
+    public ContactList() {
+        this.contactList = new ArrayList<Contact>();
+    }
+
+    public ContactList(ArrayList<Contact> contactList) {
+        this.contactList = contactList;
+    }
+
+    /**
+     * Returns true if list is modified, else false.
+     */
+    public boolean getIsModified() {
+        return isModified;
+    }
+
+    /**
+     * Sets status of whether list is modified.
+     */
+    public void setModified(boolean modified) {
+        isModified = modified;
+    }
+
+    /**
+     * Add the given contact into the contactList.
+     *
+     * @param contact The contact to be added.
+     */
+    public void addContact(Contact contact) {
+        contact.setId(getLargestId(this.contactList) + 1);
+        contactList.add(contact);
+    }
+
+    /**
+     * Retrieve the contact at the given index of the contactList.
+     *
+     * @param id The id of the contact in the contactList.
+     * @return The contact object with the specified id.
+     */
+    public Contact getContact(int id) {
+        return contactList.get(id);
+    }
+
+    /**
+     * Modify the value of the contact at the given index in the contactList.
+     *
+     * @param id      The id of the contact in the list.
+     * @param contact The contact with modified values.
+     */
+    public void editContact(int id, Contact contact) {
+        contactList.set(id, contact);
+    }
+
+    /**
+     * Removes the contact at the given index of the contactList.
+     *
+     * @param id The id of the contact in the list.
+     */
+    public Contact deleteContact(int id) {
+        int index = findIndexWithId(id);
+        if (index >= 0) {
+            Contact expense = getContact(index);
+            contactList.remove(index);
+            return expense;
+        }
+        return null;
+    }
+
+    /**
+     * Get the current number of contacts in the contactList.
+     *
+     * @return The number of contacts in the list.
+     */
```

Comment:
Trivial methods do not need header comments. There is an issue of long classes where there are more comments than actual code in this project.